### PR TITLE
Pin pip at 9.0.1 (currently most recent) in our own uses

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
Implements a small work item requested in the code review comments of [pull request 11595](https://github.com/grpc/grpc/pull/11595).

### CODE REVIEW

File: templates/tools/dockerfile/python_deps.include [hunk 1]
```diff
@@ -9,6 +9,6 @@ RUN apt-get update && apt-get install -y ${'\\'}
     python-pip
 
 # Install Python packages from PyPI
-RUN pip install pip --upgrade
+RUN pip install --upgrade pip==9.0.1
```
Comment (mehrdada) [inline templates/tools/dockerfile/python_deps.include]: What is the behavior of `--upgrade` if the existing version is `>9.0.1`?

Comment (nathanielmanistaatgoogle) [inline templates/tools/dockerfile/python_deps.include]: Downgrade. We can't test with `>9.0.1` today, but:
```
$ huh/bin/pip install --upgrade pip==9.0.1
Collecting pip==9.0.1
  Using cached pip-9.0.1-py2.py3-none-any.whl
Installing collected packages: pip
  Found existing installation: pip 7.1.2
    Uninstalling pip-7.1.2:
      Successfully uninstalled pip-7.1.2
Successfully installed pip-9.0.1
$ huh/bin/pip install --upgrade pip==8.0.0
Collecting pip==8.0.0
  Downloading pip-8.0.0-py2.py3-none-any.whl (1.2MB)
    100% |████████████████████████████████| 1.2MB 991kB/s 
Installing collected packages: pip
  Found existing installation: pip 9.0.1
    Uninstalling pip-9.0.1:
      Successfully uninstalled pip-9.0.1
Successfully installed pip-8.0.0
You are using pip version 8.0.0, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ 
```
.

**Comment by nathanielmanistaatgoogle** [comment]
[Agreed.](https://github.com/grpc/grpc/issues/11601)

**Comment by nathanielmanistaatgoogle** [comment]
Marking p1 because while this has been hanging out waiting for passing tests runs underlying changes have introduced conflicts, so it's the kind of thing that if we don't actually push it through with some sense of priority, it will hang out unintegrated forever.

**Comment by matt-kwong** [comment]
Currently uploading the images to Dockerhub. It takes a while because of multiple Dockerfiles being changed, but I'll trigger tests as soon as I'm done uploading.

**Comment by grpc-testing** [comment]
```
[microbenchmarks] Performance differences noted:
Benchmark                                                               cpu_time    real_time
----------------------------------------------------------------------  ----------  -----------
BM_PumpStreamServerToClient<InProcess>/2M                               +4%         +4%
BM_StreamingPingPong<InProcessCHTTP2, NoOpMutator, NoOpMutator>/128M/2              +5%
BM_UnaryPingPong<InProcessCHTTP2, NoOpMutator, NoOpMutator>/0/128M                  +4%
BM_UnaryPingPong<TCP, NoOpMutator, NoOpMutator>/0/2M                    +5%         +14%
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_trickle.BM_PumpStreamServerToClient_Trickle_2M_32k.opt.old: 1


[trickle] No significant performance differences
```

**Comment by matt-kwong** [comment]
Tests have finished, and I don't see any Docker-related issues in the results.

**Pull request merged**